code/modules/modular_computers/file_system/programs/blackjack.dm

```diff
@@ -0,0 +1,266 @@
+
+	var/game_id = 0
+
+	var/datum/bank_account/temp_account // Call refresh_id_account() or get_credit_amount() before using this
+
+/datum/computer_file/program/blackjack/proc/refresh_id_account()
+	temp_account = null
+	if(holder && istype(holder.holder, /obj/item/modular_computer))
+		var/obj/item/modular_computer/computer = holder.holder
+		var/obj/item/card/id/stored_card = computer.GetID()
+		if(istype(stored_card) && istype(stored_card.registered_account))
+			temp_account = stored_card.registered_account
+			return stored_card.registered_account
+
+/datum/computer_file/program/blackjack/proc/get_credit_amount()
+	refresh_id_account()
+	return temp_account?.account_balance || credits_stored
+
+/datum/computer_file/program/blackjack/proc/adjust_credits(amt)
+	if(refresh_id_account())
+		temp_account.adjust_money(amt)
+	else
+		credits_stored += amt
+
+/datum/computer_file/program/blackjack/proc/new_game()
+	if(set_wager <= 0 || get_credit_amount() < set_wager)
+		return
+
+	game_id++
+
+	adjust_credits(-set_wager)
+	active_wager = set_wager
+
+	game_state = BLACKJACK_CONTINUE
+	game_end_reason = ""
+	active_player_cards = list()
+	active_dealer_cards = list()
+	current_deck = list(
+		"2S","3S","4S","5S","6S","7S","8S","9S","10S","JS","QS","KS","AS",
+		"2H","3H","4H","5H","6H","7H","8H","9H","10H","JH","QH","KH","AH",
+		"2C","3C","4C","5C","6C","7C","8C","9C","10C","JC","QC","KC","AC",
+		"2D","3D","4D","5D","6D","7D","8D","9D","10D","JD","QD","KD","AD"
+	)
+	shuffle_inplace(current_deck)
+
+	// smooth
+	play_snd('sound/machines/switch3.ogg')
+	active_player_cards += pop(current_deck)
+	active_dealer_cards += pop(current_deck)
+	active_player_cards += pop(current_deck)
+
+/datum/computer_file/program/blackjack/proc/calculate_hand_worth(list/hand)
+	var/worth = 0
+	var/aces = 0
+	var/list/hand_copy = LAZYCOPY(hand)
+	for(var/card in hand_copy)
+		card = copytext(card, 1, -1) // Remove the suit, we do not care about it
+		if(text2num(card))
+			worth += text2num(card)
+		else
+			switch(card)
+				if("J", "Q", "K") // "Picture" cards are worth 10
+					worth += 10
+				if("A") // Aces are calculated last
+					aces += 1
+
+	for(var/i in 1 to aces) // Ace is worth either 1 or 11, whichever is more beneficial
+		if(worth < 12 - aces) // This algorithm will never surpass 21 unless you have too many aces (somehow)
+			worth += 11
+		else
+			worth += 1
+
+	return worth
+
+/datum/computer_file/program/blackjack/ui_data(mob/user)
+	var/list/data = get_header_data()
+
+	data["game_state"] = game_state
+	data["active_dealer_cards"] = active_dealer_cards
+	data["active_player_cards"] = active_player_cards
+	data["credits_stored"] = get_credit_amount()
+	data["local_credits"] = credits_stored
+	data["has_id"] = !!temp_account
+	data["active_wager"] = active_wager
+	data["set_wager"] = set_wager
+	data["game_end_reason"] = game_end_reason
+
+	return data
+
+/datum/computer_file/program/blackjack/try_insert(obj/item/I, mob/living/user)
```

Comment:
can you remove the single-letter var here

Reply:
Yes